libs/frontend/util-tooltip/src/lib/use-tooltip.ts

```diff
@@ -50,6 +52,9 @@ export const useTooltip = (options?: IUseTooltipOptions): IUseTooltip => {
   if (options?.shift) {
     middleware.push(shift())
   }
+  if (options?.flip) {
+    middleware.push(flip())
```

Comment:
I added a `flip` option that uses the [flip](https://floating-ui.com/docs/flip) middleware so that tooltip would flip up/down when needed.
However, the tooltips (edit/copy) inside the LinkTooltip will cause a "temporary" overflow for some reason.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/cd75153f-aede-4fc3-860b-f9ab8d0f146c

It may not be obvious, but at the beginning there's a small horizontal overflow happening in the builder. After swiping left and right with touchpad for a couple times, the overflow just vanished all of a sudden! Vertical swipes (that is, scroll up and down) didn't have the same effect.

I'm not 100% sure why this is happening; maybe those horizontal swipes triggered the auto-update behavior we wrote in `use-tooltip` composable.

The [autoPlacement](https://floating-ui.com/docs/autoplacement) middleware did fix this, but it also led to another problem - the edit&copy tooltip is now always positioned to the left/right side of the item. We also have to manually deal with the position&rotation of the arrow.

![Screenshot 2023-12-12 at 5 39 00 PM](https://github.com/samatechtw/pubstudio-builder/assets/59676941/81d94da4-b4ff-44d8-a41a-c69eb3b25bb8)

I think this happens because the component ("Pricing" link) is the reference element of these tooltips, and there was't enough vertical space for the tooltips, so it just went horizontal.

It would be great if you could take a look at this, sorry about that! 😵

Reply:
I see what you mean! But I'm not able to reliably reproduce it 🤔 

I think the behavior is good enough for now, I don't see it causing a big problem.